cli/contract.go

```diff
@@ -0,0 +1,157 @@
+	var cmd = &cobra.Command{
+		Use:   "contract",
+		Short: "Manage keploy contracts",
+	}
+
+	cmd.AddCommand(Generate(ctx, logger, serviceFactory, cmdConfigurator, cmd.Name()))
+	cmd.AddCommand(Download(ctx, logger, serviceFactory, cmdConfigurator, cmd.Name()))
+	cmd.AddCommand(Validate(ctx, logger, serviceFactory, cmdConfigurator, cmd.Name()))
+
+	return cmd
+}
+
+func Generate(ctx context.Context, logger *zap.Logger, serviceFactory ServiceFactory, cmdConfigurator CmdConfigurator, parentCmd string) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:     "generate",
+		Short:   "Generate contract for specified services",
+		Example: `keploy contract generate --service="email,notify"`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return cmdConfigurator.Validate(ctx, cmd)
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			svc, err := serviceFactory.GetService(ctx, parentCmd)
+			if err != nil {
+				utils.LogError(logger, err, "failed to get service")
+				return nil
+			}
+			var contract contractSvc.Service
+			var ok bool
+			if contract, ok = svc.(contractSvc.Service); !ok {
+				utils.LogError(logger, nil, "service doesn't satisfy contract service interface")
+				return nil
+			}
+			// Extract services from the flag
+			serviceStr, _ := cmd.Flags().GetStringSlice("services")
+			testStr, _ := cmd.Flags().GetStringSlice("tests")
+			var genAllMocks bool = true
+			var genAllTests bool = true
+
+			if len(serviceStr) != 0 {
+				genAllMocks = false
+			}
+			if len(testStr) != 0 {
+				genAllTests = false
+			}
+			err = contract.Generate(ctx, genAllTests, genAllMocks)
+
+			if err != nil {
+				utils.LogError(logger, err, "failed to generate contract")
+				return nil
+			}
+
+			return nil
+		},
+	}
+
+	err := cmdConfigurator.AddFlags(cmd)
+	if err != nil {
+		utils.LogError(logger, err, "failed to add generate flags")
+		return nil
+	}
+
+	return cmd
+}
+
+func Download(ctx context.Context, logger *zap.Logger, serviceFactory ServiceFactory, cmdConfigurator CmdConfigurator, parentCmd string) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:     "download",
+		Short:   "Download contract for specified services",
+		Example: `keploy contract download --service="email,notify" --path /local/path`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return cmdConfigurator.Validate(ctx, cmd)
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			svc, err := serviceFactory.GetService(ctx, parentCmd)
+			if err != nil {
+				utils.LogError(logger, err, "failed to get service")
+				return nil
+			}
+			var contract contractSvc.Service
+			var ok bool
+			if contract, ok = svc.(contractSvc.Service); !ok {
+				utils.LogError(logger, nil, "service doesn't satisfy contract service interface")
+				return nil
+			}
+			// Extract services from the flag
```

Comment:
Here again, since you have this flag in the config file, you can decide what to do with the driven flag in the service layer.